DFC.ServiceTaxonomy.GraphSync/Managers/CustomContentDefinitionManager.cs

```diff
@@ -0,0 +1,158 @@
+        }
+
+        public ContentTypeDefinition GetTypeDefinition(string name)
+        {
+            //Bypass caching as caches are scoped
+            return Build(LoadContentDefinitionRecord().ContentTypeDefinitionRecords.FirstOrDefault(x => x.Name.Equals(name, System.StringComparison.CurrentCultureIgnoreCase)), LoadContentDefinitionRecord().ContentPartDefinitionRecords);
+        }
+
+        public async Task<int> GetTypesHashAsync()
+        {
+            return await _contentDefinitionManager.GetTypesHashAsync();
+        }
+
+        public IEnumerable<ContentPartDefinition> ListPartDefinitions()
+        {
+            return _contentDefinitionManager.ListPartDefinitions();
+        }
+
+        public IEnumerable<ContentTypeDefinition> ListTypeDefinitions()
+        {
+            return _contentDefinitionManager.ListTypeDefinitions();
+        }
+
+        public ContentPartDefinition LoadPartDefinition(string name)
+        {
+            //Bypass caching as caches are scoped
+            return Build(LoadContentDefinitionRecord().ContentPartDefinitionRecords.FirstOrDefault(x => x.Name.Equals(name, System.StringComparison.CurrentCultureIgnoreCase)));
+        }
+
+        public IEnumerable<ContentPartDefinition> LoadPartDefinitions()
+        {
+            return _contentDefinitionManager.LoadPartDefinitions();
+        }
+
+        public ContentTypeDefinition LoadTypeDefinition(string name)
+        {
+            //Bypass caching as caches are scoped
+            return Build(LoadContentDefinitionRecord().ContentTypeDefinitionRecords.FirstOrDefault(x => x.Name.Equals(name, System.StringComparison.CurrentCultureIgnoreCase)), LoadContentDefinitionRecord().ContentPartDefinitionRecords);
+        }
+
+        public IEnumerable<ContentTypeDefinition> LoadTypeDefinitions()
+        {
+            return _contentDefinitionManager.LoadTypeDefinitions();
+        }
+
+        public void StorePartDefinition(ContentPartDefinition contentPartDefinition)
+        {
+            _contentDefinitionManager.StorePartDefinition(contentPartDefinition);
+        }
+
+        public void StoreTypeDefinition(ContentTypeDefinition contentTypeDefinition)
+        {
+            _contentDefinitionManager.StoreTypeDefinition(contentTypeDefinition);
+        }
+
+        /// <summary>
+        /// Returns the document from the store to be updated.
+        /// </summary>
+        private ContentDefinitionRecord LoadContentDefinitionRecord() =>
+            _contentDefinitionStore.LoadContentDefinitionAsync().GetAwaiter().GetResult();
+
+        private ContentTypeDefinition Build(ContentTypeDefinitionRecord source, IList<ContentPartDefinitionRecord> partDefinitionRecords)
+        {
+            if (source == null)
+            {
+#pragma warning disable CS8603 // Possible null reference return.
+                return null;
```

Comment:
I think the warning is valid here. The method returns a non-nullable reference type, but the method returns null.
As the method is private, we should be able to return `ContentTypeDefinition?`.